src/ngraph/op/softmax.cpp

```diff
@@ -91,3 +92,61 @@ void op::Softmax::generate_adjoints(autodiff::Adjoints& adjoints, const NodeVect
     auto x = input(0).get_source_output();
     adjoints.add_delta(x, adjoint);
 }
+
+// *** SOFTMAX OP SET 1 ***
+const string op::set1::Softmax::type_name{"Softmax"};
+
+op::set1::Softmax::Softmax(const Output<Node>& arg, const size_t axis)
+    : UnaryElementwiseArithmetic(arg)
+    , m_axis(axis)
+{
+    set_opset_version(1);
+    constructor_validate_and_infer_types();
+
+    NODE_VALIDATION_CHECK(this,
+                          axis >= 0 && axis < get_shape().size(),
+                          "Reduction axis (",
+                          axis,
+                          ") is out of bounds (argument shape: ",
+                          get_shape(),
+                          ").");
+}
+
+shared_ptr<Node> op::set1::Softmax::copy_with_new_args(const NodeVector& new_args) const
+{
+    check_new_args_count(this, new_args);
+    return make_shared<op::set1::Softmax>(new_args.at(0), m_axis);
+}
+
+void op::set1::Softmax::generate_adjoints(autodiff::Adjoints& adjoints, const NodeVector& deltas)
+{
+    auto delta = deltas.at(0);
+
+    auto z = delta * shared_from_this();
+
+    std::vector<size_t> axes(get_shape().size() - m_axis);
+    std::iota(std::begin(axes), std::end(axes), m_axis);
```

Comment:
This algorithm looks little suspicious to me.
Do we have any unit test to verify it?

Reply:
Can you clarify your doubts?
The place where `generate_adjoints` can be verified is `ngraph/test/backend/autodiff.in.cpp`, but any backend has not supported `v1` opset version yet.
Look at:
```
size_t opset_version = node.get_opset_version();
        if (opset_version != 0)
```
in `int_executable.hpp `
Have you idea to test it?

Reply:
@diyessi What do you think about this part of algorithm? Have you maybe idea how to test `generate_adjoints`? ;)

Reply:
Ok, thanks ;) @aprocter Should we throw `Not supported yet` exception in such case or is it acceptable to keep current version?